### ISSUE TITLE
How to train for paddleOCR for new Language(Bengali or Bangla)

### ISSUE THREAD

**Issue by iambechitra** [PaddlePaddle/PaddleOCR#5799]
I want to train paddleOCR for Bengali+English Language. But couldn't find any tutorial or reference to do so. How can i train PaddleOCR for Bangali+English Language Together. 

I found a Bengali Vocab Corpora(60000+ words) in the given link bellow
https://raw.githubusercontent.com/JaidedAI/EasyOCR/master/easyocr/dict/bn.txt

also the list of Bengali latter in the given link bellow -
https://raw.githubusercontent.com/JaidedAI/EasyOCR/master/easyocr/character/bn_char.txt

Since the Open Source EasyOCR library already has the required files to train for Bengali Language in PaddleOCR so i was thinking whether is it possible to train PaddleOCR for Bengali Language. 

I want to make paddleOCR work for both Bengali and English Language Together similar to Chinese and English language. 
Please guide me how to do that.

**Comment by felix115** [comment]
There is not much information. I guess something may be wrong in dataset processing.check dataset.py.

**Comment by iambechitra** [comment]
Thanks for the reply. I was able to train the model by modifying the YAML file. But as of now i'm getting little low accuracy. Can you guide me how to increase the accuracy? 

Also guide me how to add dictionary(language corpus like 50000+ words as an addition) while training?
In current YAML file i can add letter as Bengali Language not Words so how can i do that?

**Comment by letsyookyung** [comment]
## train configs file 
I made customized training files, and it worked well. so please check if you would like to refer to it. 
(Please ignore any korean in my file, I just didn't change anything from original training configs file)
  

#### det 
[https://github.com/letsyookyung/TIL/blob/main/customized_det.yml](url)

#### rec (korean)
[https://github.com/letsyookyung/TIL/blob/main/customized_rec.yml](url)


## rec dictionary 
I guess it is character(?) based recognition, so you don't need to add any words in dictionary. 
each character of word should be added in the dictionary without reduplication, if you didn't add then the model can not infer that character. I will also share my korean dictionary form. I used pretrained model so my new korean dictionary includes chinese character, just leave it and just add characters which you used in train data. 

[https://github.com/letsyookyung/TIL/blob/main/new_korean_dict.txt](url)

**Comment by mobassir94** [comment]
@iambechitra how did you solve the accuracy = 0.000 issue? can you please share the updated YAML file?

**Comment by bit-scientist** [comment]
@letsyookyung hello, your new_korean_dict.txt seems to contain both Japanese (I could find these 6: アカ, グ, ニ, ラ, ン) and Chinese (before and after main Korean characters) characters. I have some questions, if you don't mind answering 😄 

1. Could you tell me why you added these particular Japanese characters only?
2. What is the reason to add Chinese characters in Korean dictionary?
3.  Why did you separate Chinese characters into two locations (mainly before Korean letters and some after Korean letters)? Are they different hanja characters that are used in Korean society too?
4. What type of hanja characters are used in Korean official documents like invitations, ID card names, etc? I would love it, if you could provide a full hanja dictionary used in Korea.
Thank you in advance.

**Comment by dorsamotiallah** [comment]
Hi, I have encountered the same problem. could you please share your final version of the YAML file? (after fixing accuracy = 0 problem)

**Comment by masoudMZB** [comment]
after changing the dictionary file, some layers will drop and cause bad prediction . your acc will be 0.000000 until first iteration of validation. Check your CTCLoss, and other loss, they are decreasing, let model to continue training and acc will increase soon

My question is how to freeze some layers ?

**Comment by MeherajUlMahmmud** [comment]
@iambechitra Could you please provide the image dataset for model training?

**Comment by rm-asif-amin** [comment]
@masoudMZB I'm trying to train for the same language(it's not supported by paddle yet). Here's my approach and outcome-
1. Training a model from scratch with the following yml config(click to expand) -
<details>
Global:
  use_gpu: true
  epoch_num: 10
  log_smooth_window: 20
  print_batch_step: 10
  save_model_dir: ./output/rec_ben
  save_epoch_step: 3
  eval_batch_step:
  - 0
  - 100
  cal_metric_during_train: true
  pretrained_model: null
  checkpoints: null
  save_inference_dir: /content/inference
  use_visualdl: false
  infer_img: 
  character_dict_path: /content/bn_dict.txt
  max_text_length: 25
  infer_mode: false
  use_space_char: true
Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  lr:
    name: Cosine
    learning_rate: 0.001
  regularizer:
    name: L2
    factor: 1.0e-05
Architecture:
  model_type: rec
  algorithm: CRNN
  Transform: null
  Backbone:
    name: MobileNetV3
    scale: 0.5
    model_name: small
    small_stride:
    - 1
    - 2
    - 2
    - 2
  Neck:
    name: SequenceEncoder
    encoder_type: rnn
    hidden_size: 48
  Head:
    name: CTCHead
    fc_decay: 1.0e-05
Loss:
  name: CTCLoss
PostProcess:
  name: CTCLabelDecode
Metric:
  name: RecMetric
  main_indicator: acc
Train:
  dataset:
    name: SimpleDataSet
    data_dir: /content/content/final_dataset/rec/train/train
    label_file_list:
    - /content/content/final_dataset/rec/train/rec_gt_train.txt
    transforms:
    - DecodeImage:
        img_mode: BGR
        channel_first: false
    - RecAug: null
    - CTCLabelEncode: null
    - RecResizeImg:
        image_shape:
        - 3
        - 32
        - 320
    - KeepKeys:
        keep_keys:
        - image
        - label
        - length
  loader:
    shuffle: true
    batch_size_per_card: 256
    drop_last: true
    num_workers: 8
Eval:
  dataset:
    name: SimpleDataSet
    data_dir: /content/content/final_dataset/rec/test/test
    label_file_list:
    - /content/content/final_dataset/rec/test/rec_gt_test.txt
    transforms:
    - DecodeImage:
        img_mode: BGR
        channel_first: false
    - CTCLabelEncode: null
    - RecResizeImg:
        image_shape:
        - 3
        - 32
        - 320
    - KeepKeys:
        keep_keys:
        - image
        - label
        - length
  loader:
    shuffle: false
    drop_last: false
    batch_size_per_card: 256
    num_workers: 8

</details>
2. Using a dataset of ~80k one word images. After training for 100+ epochs, the training loss looks like this-

![image](https://github.com/PaddlePaddle/PaddleOCR/assets/50940878/cbca8b51-f048-4fef-a0cd-f15ebff24ca0)
4. The training, eval accuracy are not coherent ( low eval accuracy compared to train)-
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/50940878/e0a62f6c-c922-48a3-9f1d-cabc7ff26001)

5. The Edit distance metrics, on the other hand are opposite -
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/50940878/b3abfaf6-a2e7-4b4b-9dfa-aa9ff84ba2a0)
6. I've added Bengali letters to an English dictionary which seems to be encoding correctly-
```
Original Label: ব্যঙ্গবিদ্রূপাত্মক
Transformed Label: [157 160 139 137 157 123 152 161 126 155 122 150 159 135   0   0   0   0 
```
My questions are-
1. Is my approach right? The model seems to be learning, why the low accuracy?
2. Should I use a pre-trained model and finetune?( A similar indic language maybe? )

Would love to know your opinion.

**Comment by rm-asif-amin** [comment]
@iambechitra Did you manage to solve this? Facing the same for Bangla
